examples/model_selection/plot_cost_sensitive_learning.py

```diff
@@ -0,0 +1,752 @@
+
+    .. [2] `Charles Elkan, "The Foundations of Cost-Sensitive Learning",
+       International joint conference on artificial intelligence.
+       Vol. 17. No. 1. Lawrence Erlbaum Associates Ltd, 2001.
+       <https://cseweb.ucsd.edu//~elkan/rescale.pdf>`_
+"""
+
+# %%
+# Cost-sensitive learning with constant gains and costs
+# -----------------------------------------------------
+#
+# In this first section, we illustrate the use of the
+# :class:`~sklearn.model_selection.TunedThresholdClassifier` in a setting of
+# cost-sensitive learning when the gains and costs associated to each entry of the
+# confusion matrix are constant. We use the problematic presented in [2]_ using the
+# "Statlog" German credit dataset [1]_.
+#
+# "Statlog" German credit dataset
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# We fetch the German credit dataset from OpenML.
+import sklearn
+from sklearn.datasets import fetch_openml
+
+sklearn.set_config(transform_output="pandas")
+
+german_credit = fetch_openml(data_id=31, as_frame=True, parser="pandas")
+X, y = german_credit.data, german_credit.target
+
+# %%
+# We check the feature types available in `X`.
+X.info()
+
+# %%
+# Many features are categorical and usually string-encoded. We need to encode
+# these categories when we develop our predictive model. Let's check the targets.
+y.value_counts()
+
+# %%
+# Another observation is that the dataset is imbalanced. We would need to be careful
+# when evaluating our predictive model and use a family of metrics that are adapted
+# to this setting.
+#
+# In addition, we observe that the target is string-encoded. Some metrics
+# (e.g. precision and recall) require to provide the label of interest also called
+# the "positive label". Here, we define that our goal is to predict whether or not
+# a sample is a "bad" credit.
+pos_label, neg_label = "bad", "good"
+
+# %%
+# To carry our analysis, we split our dataset using a single stratified split.
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, random_state=0)
+
+# %%
+# We are ready to design our predictive model and the associated evaluation strategy.
+#
+# Evaluation metrics
+# ^^^^^^^^^^^^^^^^^^
+#
+# In this section, we define a set of metrics that we use later. To see
+# the effect of tuning the cut-off point, we evaluate the predictive model using
+# the Receiver Operating Characteristic (ROC) curve and the Precision-Recall curve.
+# The values reported on these plots are therefore the true positive rate (TPR) and
+# the false positive rate (FPR) for the ROC curve and the precision and recall for the
+# Precision-Recall curve.
+#
+# From these four metrics, scikit-learn does not provide a scorer for the FPR. We
+# therefore need to define a small custom function to compute it.
+from sklearn.metrics import confusion_matrix
+
+
+def fpr_score(y, y_pred, neg_label, pos_label):
```

Comment:
should that just be a separate function? This example finally puts together a bunch of stuff that I wanted for my classes for many years, I think having this function also in the main code to complete it would be great.

Reply:
There's a PR for it: https://github.com/scikit-learn/scikit-learn/pull/19556